### PR TITLE
better ensure env to param based job updates prior to initiating job …

### PR DESCRIPTION
**Pull request by gabemontero**
…runs

Fixes https://github.com/openshift/origin/issues/18324

@openshift/sig-developer-experience fyi / ptal

### CODE REVIEW

**Comment by gabemontero** [comment]
[test]

**Comment by gabemontero** [comment]
heads up ... some formatting updates (tabs for spaces) ... try `?w=1` on the PR review

File: src/main/java/io/fabric8/jenkins/openshiftsync/JenkinsUtils.java [hunk 1]
```diff
@@ -268,125 +297,143 @@ else if (envKeys.contains(param.getName())) {
...
+            // grab envs from actual build in case user overrode default values
+            // via `oc start-build -e`
+            JenkinsPipelineBuildStrategy strat = build.getSpec().getStrategy()
+                    .getJenkinsPipelineStrategy();
+            // only add new param defs for build envs which are not in build
+            // config envs
+            Map<String, ParameterDefinition> paramMap = addJobParamForBuildEnvs(
+                    job, strat, false);
+            // TODO do we want to abort the launching of the job if
+            // verifyEnvVars returns false?
```
Comment (bparees) [inline src/main/java/io/fabric8/jenkins/openshiftsync/JenkinsUtils.java]: for now i'd say probably not, just logging it is good.

**Comment by bparees** [comment]
lgtm

File: src/main/java/io/fabric8/jenkins/openshiftsync/JenkinsUtils.java (hunk 1 above)
Comment (gabemontero) [reply src/main/java/io/fabric8/jenkins/openshiftsync/JenkinsUtils.java]: agreed - thanks

**Comment by gabemontero** [comment]
[merge]

**Pull request merged**